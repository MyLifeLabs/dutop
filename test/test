#! /bin/sh
#
# Create a sample file hierarchy and run dutop on it.
#
# Usage: ./test [path to dutop executable]
#

set -eu

dutop="${1:-dutop}"

rm -rf root
mkdir -p root
(
  # Create a tree with 3 regular files containing 1 byte each.
  cd root

  # Regular files
  echo -n x > a
  echo -n x > b

  # Symbolic link
  ln -s a a2

  # Hard link
  ln b b2

  # Sub-directory
  mkdir -p sub
  echo -n x > sub/c
)

"$dutop" root
