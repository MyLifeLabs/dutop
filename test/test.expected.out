   33.3%                     1 root/b
   33.3%                     1 root/sub/c
d  33.3%                     1 root/sub
   33.3%                     1 root/a
   33.3%                     1 root/b2 [root/b]
d 100.0%                     3 root
